### PR TITLE
[processor/resourcedetection] Config check for EKS to verify the node is on AWS

### PR DESCRIPTION
**Pull request by aateeqi**
**Description:** 
Check that an EKS resource is running on AWS by checking the if the auth config exists
adopted from https://github.com/open-telemetry/opentelemetry-go-contrib/blob/7e31ebe040306aee2c826972269f938f9f0e7c34/detectors/aws/eks/detector.go#L46

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2977

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: processor/resourcedetectionprocessor/internal/aws/eks/detector.go [hunk 1]
```diff
@@ -31,25 +35,42 @@ const (
...
+	getConfigMap(ctx context.Context, namespace string, name string) (map[string]string, error)
+}
+
+type eksDetectorUtils struct {
+	clientset *kubernetes.Clientset
+}
 
 // Detector for EKS
-type Detector struct{}
+type Detector struct {
```
Comment (Aneurysm9) [inline processor/resourcedetectionprocessor/internal/aws/eks/detector.go:51]: Does this struct need to be exported?  It seems like not.

File: processor/resourcedetectionprocessor/internal/aws/eks/detector.go [hunk 2]
```diff
@@ -31,25 +35,42 @@ const (
...
 func (detector *Detector) Detect(ctx context.Context) (resource pdata.Resource, schemaURL string, err error) {
 	res := pdata.NewResource()
 
-	// Check if running on k8s.
-	if os.Getenv(kubernetesServiceHostEnvVar) == "" {
-		return res, "", nil
+	//Check if running on EKS.
+	isEKS, err := isEKS(ctx, detector.utils)
+	if !isEKS {
+		return res, "", err
```
Comment (Aneurysm9) [inline processor/resourcedetectionprocessor/internal/aws/eks/detector.go:73]: Consider adding a debug log here indicating that the detector only works on EKS.

File: processor/resourcedetectionprocessor/internal/aws/eks/detector.go [hunk 3]
```diff
@@ -31,25 +36,44 @@ const (
...
+func (detector *detector) Detect(ctx context.Context) (resource pdata.Resource, schemaURL string, err error) {
 	res := pdata.NewResource()
 
-	// Check if running on k8s.
-	if os.Getenv(kubernetesServiceHostEnvVar) == "" {
-		return res, "", nil
+	//Check if running on EKS.
+	isEKS, err := isEKS(ctx, detector.utils)
+	if !isEKS {
+		detector.logger.Debug("EKS envi", zap.Error(err))
```
Comment (Aneurysm9) [inline processor/resourcedetectionprocessor/internal/aws/eks/detector.go:75]:
```suggestion
		detector.logger.Debug("Unable to identify EKS environment", zap.Error(err))
```

**Review by Aneurysm9 — APPROVED** [review]

**Comment by jpkrohling** [comment]
This test failed, but doesn't look related, although it's suspiciously close:

```
=== RUN   TestFactory
E0126 21:31:07.295817   53857 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 52 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x23bbd00, 0x383f740})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/runtime/runtime.go:74 +0xe6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000147898})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/runtime/runtime.go:48 +0xb0
panic({0x23bbd00, 0x383f740})
	/opt/hostedtoolcache/go/1.17.6/x64/src/runtime/panic.go:1047 +0x266
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1(0xc0001c21c0, 0xc0001b4240, 0xc0003e2000, 0xc000147b00)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.2/tools/cache/reflector.go:319 +0x127a
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch(0xc0001c21c0, 0xc0003e2000)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.2/tools/cache/reflector.go:361 +0x2d6
k8s.io/client-go/tools/cache.(*Reflector).Run.func1()
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.2/tools/cache/reflector.go:221 +0x45
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc00012f6d8)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/wait/wait.go:155 +0x82
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000060540, {0x2912920, 0xc0001122d0}, 0x1, 0xc0003e2000)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/wait/wait.go:156 +0xcf
k8s.io/client-go/tools/cache.(*Reflector).Run(0xc0001c21c0, 0xc0003e2000)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.2/tools/cache/reflector.go:220 +0x296
k8s.io/apimachinery/pkg/util/wait.(*Group).StartWithChannel.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/wait/wait.go:56 +0x3f
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/wait/wait.go:73 +0x7f
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/wait/wait.go:71 +0xdf
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference

goroutine 52 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000147898})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/runtime/runtime.go:55 +0x145
panic({0x23bbd00, 0x383f740})
	/opt/hostedtoolcache/go/1.17.6/x64/src/runtime/panic.go:1047 +0x266
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1(0xc0001c21c0, 0xc0001b4240, 0xc0003e2000, 0xc000147b00)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.2/tools/cache/reflector.go:319 +0x127a
k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch(0xc0001c21c0, 0xc0003e2000)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.2/tools/cache/reflector.go:361 +0x2d6
k8s.io/client-go/tools/cache.(*Reflector).Run.func1()
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.2/tools/cache/reflector.go:221 +0x45
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc00012f6d8)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/wait/wait.go:155 +0x82
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000060540, {0x2912920, 0xc0001122d0}, 0x1, 0xc0003e2000)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/wait/wait.go:156 +0xcf
k8s.io/client-go/tools/cache.(*Reflector).Run(0xc0001c21c0, 0xc0003e2000)
	/home/runner/go/pkg/mod/k8s.io/client-go@v0.23.2/tools/cache/reflector.go:220 +0x296
k8s.io/apimachinery/pkg/util/wait.(*Group).StartWithChannel.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/wait/wait.go:56 +0x3f
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/wait/wait.go:73 +0x7f
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.23.2/pkg/util/wait/wait.go:71 +0xdf
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver	0.120s
```

I'm retriggering the job, but could you check whether this is related? This is especially suspicious as there was a change to the go client library as part of this PR.

**Comment by aateeqi** [comment]
@jpkrohling looks like its passing after upgrading to `k8s.io/client-go v0.23.2` and running `make gotidy`

**Pull request merged**